### DISCUSSION TITLE
Module 05 Dataloader:  shuffle default value false

### DISCUSSION THREAD

**Discussion by asgalon** [harvard-edge/cs249r_book#1230] [TinyTorch Q&A] [answered]
Hi,

I have a question: Why is the default value of the DataLoader __init__() function shuffle parameter False, when below  "Common Pitfalls and Best Practices" has as first common mistake "forgetting to shuffle training data"? Shouldn't the default value align with the best practice here? What is the main use case for setting shuffle=False there?

Edit: Oh, I see... for validation normally no shuffle is needed. So on average, in half of the use cases (training) the default value is mostly wrong in the other half (validation) it matches. In that case, wouldn't it be better to replace the shuffle parameter with a boolean "isTraining" parameter without a default value? Probably not the most important of API design questions, admittedly.

Edit2: Thinking about it again, the default value should probably be True anyway, because if it is omitted, default=False can provoke overfitting in training, while default=True would only do an additional shuffling step in validation, which should not have any effects on the validation result.

Edit3: shuffle=False in validation/test for reproducible metrics. So perhaps no default value would be better, shuffle being synonymous with "isTraining". Sorry, sometimes I tend to obsess about such details....

**Comment by profvjreddi** [comment, marked as answer]
Good question @asgalon, and you actually answered it yourself! The default is False because DataLoaders are used for both training and evaluation, and evaluation should not be shuffled (you want reproducible ordering). PyTorch does the same thing. The "common pitfall" note is there precisely because it is a default that trips people up, which is the pedagogical point.
